File: app/src/main/java/com/quiz/mallzellij_show_android/UserSession.java
package com.quiz.mallzellij_show_android;

import com.quiz.mallzellij_show_android.model.AuthResponse;

public class UserSession {
    private static UserSession instance;
    private Long id;
    private String name;
    private String email;
    private String role;

    private UserSession() {}

    public static UserSession getInstance() {
        if (instance == null) {
            instance = new UserSession();
        }
        return instance;
    }

    public void login(AuthResponse response) {
        this.id = response.getId();
        this.name = response.getName();
        this.email = response.getEmail();
        this.role = response.getRole();
    }

    public void logout() {
        this.id = null;
        this.name = null;
        this.email = null;
        this.role = null;
    }

    public boolean isAdmin() {
        return "admin".equalsIgnoreCase(role);
    }

    public String getRole() { return role; }
    public String getEmail() { return email; }
}

File: app_flutter/lib/services/user_session.dart
import 'package:shared_preferences/shared_preferences.dart';
import '../models/auth_response.dart';

class UserSession {
  static final UserSession _instance = UserSession._();
  factory UserSession() => _instance;
  UserSession._();

  int? _id;
  String? _name;
  String? _email;
  String? _role;

  int? get id => _id;
  String? get name => _name;
  String? get email => _email;
  String? get role => _role;
  bool get isAdmin => _role?.toLowerCase() == 'admin';

  void login(AuthResponse r) {
    _id = r.id;
    _name = r.name;
    _email = r.email;
    _role = r.role;
  }

  void logout() {
    _id = null;
    _name = null;
    _email = null;
    _role = null;
  }
}
